src/Illuminate/Database/Eloquent/Factories/Factory.php

```diff
@@ -182,6 +182,10 @@ public function configure()
      */
     public function raw($attributes = [], ?Model $parent = null)
     {
+        if ($this->count instanceof Closure) {
+            $this->count = call_user_func($this->count, $parent);
```

Comment:
There is a helper for this:
```php
$this->count =value($this->count, $parent);
```
This replaces the whole if block.

Reply:
Neat. Fixed. Thanks.